gremlin-dotnet/src/Gremlin.Net/Process/Traversal/DefaultTraversal.cs

```diff
@@ -82,21 +82,22 @@ public bool MoveNext()
         }
         
         private bool MoveNextInternal()
-        {                   
+        {
             if (_fetchedNext) return _nextAvailable;
+
+            if (!_nextAvailable || _nextAvailable && TraverserEnumerator.Current?.Bulk == 0)
```

Comment:
can be simplified
```suggestion
            if (!_nextAvailable || TraverserEnumerator.Current?.Bulk == 0)
```

Reply:
This code will also throw an exception if called without first starting the iterator: You will need to track that the iterator has not been started.

```.csharp
private object GetCurrent()
{
    // Use dynamic to object to prevent runtime dynamic conversion evaluation
    return TraverserEnumerator.Current?.Object;
}
```

Please review the new [.Net code here](https://source.dot.net/#PresentationFramework/MS/Internal/WindowsRuntime/Generated/WinRT/Projections/IEnumerable.cs,239)